### ISSUE TITLE
Self-hosted CartoDB

### ISSUE THREAD

**Issue by mojodna** [AmericanRedCross/posm#215]
Having a full-blown, offline-enabled CartoDB instance would be a nice addition to POSM. I'm not sure how it would be used, but I can imagine it helping @vdeparday et al.

/cc @jjmata ;-)

**Comment by jjmata** [comment]
Hehe!  Would love to help as well.  I started follow POSM exactly for that reason ... I have a `docker-compose.yml` somewhere you guys would be interested in (mixes Carto + ODM).  Who do I talk to to volunteer/figure out what makes sense?

**Comment by mojodna** [comment]
Me / here, I suppose, esp. so others can weigh in.

To do it, you'd create a `dronedb-deploy.sh` (or `cartodb-deploy.sh`) in [posm-build](https://github.com/AmericanRedCross/posm-build), similar to [`osm-deploy.sh`](https://github.com/AmericanRedCross/posm-build/blob/master/kickstart/scripts/osm-deploy.sh), except that it'd probably be simpler (less installation, probably mostly `docker fetch`, initializing a Postgres database and setting up the upstart config and Nginx proxies).

Consider PostGIS part of "core" (i.e. assume it's present and don't bring your own).

You'll need AmericanRedCross/posm-build#62 in your working branch, though I'm planning on merging that in the next couple days along with the rest.

**Comment by jjmata** [comment]
First and foremost, let's understand the use cases.  Is that where @vdeparday and others weigh in?  Hard to understand what is needed/expected until we hear from the actual users.

I'm envisioning some sort of "default processing" workflow myself (stitching, tiling, etc.) but what should be included in that list and what comes after?

**Comment by dalekunce** [comment]
I'd love to have a session on POSM with @vdeparday, HOTOSM, and other interested folks.

**Comment by mojodna** [comment]
@jjmata agreed, though I guess my thinking was that providing an out-of-the-box CartoDB instance would be suitably generic that unexpected use-cases might arise.

Re: the default processing workflows you're thinking of, those are probably better considered as part of the POSM/OpenDroneMap integration effort (which means that I should spin up your Docker image and see the mods that you've applied).

**Comment by jjmata** [comment]
Haven't actually applied any valuable changes yet, most of my work is still conceptual and lives in local branches too embarrassing to push anywhere ;-)

I agree 100% that once you have the Carto builder/editor in your hands, use cases will probably "arise" as you say.  Would be good to find a suitable/stable cut of their codebase and package it well (as it is a moving target and can be tricky to Dockerize well - speaking from experience!)  I can ask @javisantana and see what he recommends.

/cc @saleiva in case he wants to chime in as well.

**Comment by mojodna** [comment]
Dockerized version (haven't tried it out myself): https://github.com/chriswhong/docker-cartodb
